fix: add option `ignoreTrailingSlash` to MockAgent and `.intercept()`

Uzlopak opened the pull request:
fixes #3649 

Normalize the path in the mock interceptor

Uzlopak commented:
@metcoder95 

Should we wait for a second opinion/approval?

KhafraDev reviewed (changes requested):
I don't think this is a good idea.

metcoder95 commented:
@KhafraDev can you elaborate on that?

Uzlopak commented:
@metcoder95 

I chattet with @KhafraDev and he sent me following tweet of @mcollina 

https://x.com/matteocollina/status/1828802494832705896

Seems like to every issue there is a tweet by matteo :D.

Anyhow:
I reworked on this and added ignoreTrailingSlashes as an option of MockAgent. Where do we document stuff like this?

KhafraDev reviewed (commented):
Why have it as an option on the client rather than passed in the .intercept options?

metcoder95 reviewed (commented):
lgtm, here we can update the docs: https://github.com/nodejs/undici/blob/main/docs/docs/api/MockPool.md#mockpoolinterceptoptions

Uzlopak commented on lib/mock/mock-utils.js:
```diff
@@ -296,7 +320,7 @@ function mockDispatch (opts, handler) {
   }
 
-  function resume () {}
+  function resume () { }
```
```suggestion
  function resume () {}
```

mcollina reviewed (approved):
lgtm

metcoder95 commented:
Shall we backport it?
